src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaFactory.java

```diff
@@ -29,4 +31,8 @@ public NakadiKafkaConsumer createNakadiConsumer(final String topic, final List<K
         return new NakadiKafkaConsumer(getConsumer(), topic, kafkaCursors, pollTimeout);
     }
 
+    @Override
+    public void updateProperties(final Properties kafkaProperties) {
+        kafkaProducer = new KafkaProducer<>(kafkaProperties);
```

Comment:
From my understanding:

1. I think you're right. The reference to the old producer would not be update in the `KafkaTopicRepository`. Then, even if we do have a new producer here, it would not be used. I think we need to change the repository to always ask for this "producer singleton". Or we need a "setter" in the repository in order to update the reference to the new producer.

2 and 3. This seems to be too much hypothetical. But since it's as simple as locking, I would not be against it. That could prevent future developers from expecting this method to provide insufficient thread safety.